Magma 5 g

Kaleem-Wavelabs opened the pull request:
<!--
    Tag your PR title with the components that it touches.
    E.g. "[lte][agw] Changeset" or "[orc8r][docker] ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

themarwhal committed:
Add CircleCI jobs to run AGW C/C++ unit tests (#5848)

Signed-off-by: Marie Bremner <marwhal@fb.com>

themarwhal committed:
[SessionD] Add a new field to track per-rule versions (#5779)

ulaskozat committed:
[5G: MVC][component: agw][type: enhancement] changes related to AMF-PR #4595 (#5735) (#5908)

* changes related to AMF-PR #4595
Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

Co-authored-by: chandra <59682018+chandra-77@users.noreply.github.com>

gurrapualt committed:
Sessiond module MVC phase converged core changes (#3898)

Co-authored-by: Matthew Mosesohn <matthew.mosesohn@gmail.com>
Co-authored-by: Venukumar Gurrapu <venukumar.g@acldigital.com>

themarwhal committed:
[SessionD] Update rule versions on install/uninstalling rules (#5787)

gurrapualt committed:
Fixing 2nd TEID onwards missmatch to AMF (#5994)

Signed-off-by: Venu Kumar Gurrapu <venukumar.g@altencalsoftlabs.com>

ashish-pr committed:
[5G: MVC][component: agw][type: enhancement] NGAP module development #2036 NGAP Handlers Phase1 (#3569)

* Contains NGAP handlers for following messages

1.InitialContextSetup
2.initialUEMessage
3.uplinkNASTransport
4.NGSetup

Signed-off-by: ashish-pr <ashish.p@altencalsoftlabs.com>

ashish-pr committed:
Contains NGAP handlers for following messages (#6060)

* Contains NGAP handlers for following messages
    1> NASNonDeliveryIndication
    2> UEContextReleaseRequest
    3> UEContextReleaseComplete
    4> UEContextReleaseCommand
    5> PduSessionResourceReleaseRequest
    6> PduSessionResourceReleaseResponse
    7> PduSessionResourceSetupRequest

Signed-off-by: ashish-pr <ashish.p@altencalsoftlabs.com>

chandra-77 committed:
[5G: MVC][component: agw][type: enhancement] Amf core dependent files (#5989)

* AMF-CORE PR's dependent files

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

chandra-77 committed:
[5G: MVC][component: agw][type: enhancement] AMF-Core (#5984)

* AMF-Core

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

ashish-pr committed:
[5G: MVC][component: agw][type: enhancement] NGAP module development #2036 Parsing config file (#6095)

* Contains the changes to parse and populate the following fields from mme.conf
1.AMF pointer
2.AMF region id
3.AMF set id
4.mcc and mnc
5.TAC list

Signed-off-by: ashish-pr <ashish.p@altencalsoftlabs.com>

chandra-77 committed:
[5G: MVC][component: agw][type: enhancement] IDLE mode support (#5993)

* rebased with latest magma-5g

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

chandra-77 committed:
[5G: MVC][component: agw][type: enhancement] AMF-CORE with gTEID support (#5996)

* AMF-CORE with gTEID support

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

hkshiv1 committed:
[5G: MVC][component: agw][type: enhancement] Paging feature (#6024)

* Paging feature

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

hkshiv1 committed:
[5G: MVC][component: agw][type: enhancement] Periodic registration (#6025)

* PERIODIC REGISTRATION Feature Changes

Signed-off-by: hkshiv1 <shivakumar.hk@acldigital.com>

hkshiv1 committed:
[5G: MVC][component: agw][type: enhancement] IDLE Mode Feature Extension (#6030)

* IDLEMODE FEATURE Changes

Signed-off-by: hkshiv1 <shivakumar.hk@acldigital.com>

srinivas-47 committed:
[5G: MVC][component: agw][type: enhancement] Multiple UE and multiple PDU session Support (#6033)

* multi UE and multi PDU feature support

Signed-off-by: srinivas-47 <petetisrinivas47@gmail.com>

ashish-acl committed:
[5G: MVC][component: agw][type: enhancement] Initial Context Setup Request Implementation in AMF  (#6064)

* Intial Context Setup Request support for AMF

Signed-off-by: ashish-acl <ashish.t@altencalsoftlabs.com>

hkshiv1 committed:
[5G: MVC][component: agw][type: enhancement]Handling Service Request Response on Paging (#6062)

* SERVICE REQUEST FEATURE Changes

Signed-off-by: hkshiv1 <shivakumar.hk@acldigital.com>

ashish-acl committed:
[5G: MVC][component: agw][type: enhancement] Implemented the optional but conditional IEs in Registration Accept message from AMF (#6067)

* adding conditional NAS IEs to Registration Accept

Signed-off-by: ashish-acl <ashish.t@altencalsoftlabs.com>

ashish-acl committed:
[5G: MVC][component: agw][type: enhancement] Replaying UE Security capability from AMF (#6065)

* replaying UE security capability during Security Mode Command

Signed-off-by: ashish-acl <ashish.t@altencalsoftlabs.com>

rupa455 committed:
Scenario files for dsTest (#6036)

* Scenario files for dsTest

Signed-off-by: rupa455 <rupa.d@altencalsoftlabs.com>

ashish-acl committed:
[5G: MVC][component: agw][type: enhancement] NAS Security header support for AMF  (#6029)

* Added encode and decode support for Security Headers for NAS messagesin AMF

Signed-off-by: ashish-acl <ashish.t@altencalsoftlabs.com>

srinivas-47 committed:
[5G: MVC][component: agw][type: enhancement] AMF state machine feature support (#6068)

* AMF State machine Feature Support

Signed-off-by: srinivas-47 <petetisrinivas47@gmail.com>

hkshiv1 committed:
[5G: MVC][component: agw][type: enhancement]  AMF Base Timers (#6066)

* AMF Base Timers

Signed-off-by: hkshiv1 <shivakumar.hk@acldigital.com>

ashish-pr committed:
Added missing code (#6243)

Signed-off-by: ashish-pr <ashish.p@altencalsoftlabs.com>

themarwhal committed:
[PipelineD][SessionD] Cleanup flows for non-existent sessions (#6364)

Signed-off-by: Marie Bremner <marwhal@fb.com>

pshelar committed:
AGW: s1ap: skip HE test due to CI failure. (#6411)

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

prabinakpattnaik committed:
[5G: MVC][pipelined]Paging Functionality Support For 5G in pipelined (#3535)

Adds paging functionality without buffering support for 5G in pipelined.

Commit log:

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* [5G: MVC][component: agw][type: enhancement][pipelined]Paging Functionality Support For 5G in pipelined.

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

tmdzk committed:
[AGW][CI]Build 5G on its branch and push to 5g registry (#6850)

* Build 5G on its branch and push to 5g registry

Signed-off-by: Timothée Dzik <timdzik@fb.com>

* Build 5G on its branch and push to 5g registry

Signed-off-by: Timothée Dzik <timdzik@fb.com>

tmdzk committed:
Bump eventlet to fix ALREADY_HANDLED issue

Signed-off-by: Timothée Dzik <timdzik@fb.com>

ashish-acl committed:
[5G: MVC][component: agw][type: enhancement] Magma-5G merge issue fixes (#6752)

* fixes and workaround for bugs found during IT testing og magma-5g branch

Signed-off-by: ashish-acl <ashish.t@altencalsoftlabs.com>

tmdzk committed:
[CI]Remove determinators on magma-5G

Signed-off-by: Timothée Dzik <timdzik@fb.com>

tmdzk committed:
Remove flaky test

Signed-off-by: Timothée Dzik <timdzik@fb.com>

panyogesh committed:
[WIP][component: agw] PDU Session Establishment (#6972)

Fixes included :
1. NGAP & AMF Startup procedure.
2. PDU Session Establishment.
3. Unit testcases in ngap.

Testcase :
1. Unit Testcases of ngap
2. Testing using UERANSIM
3. Testing with DSTester

Signed-off-by: YOGESH PANDEY <yogesh@wavelabs.ai>

VinayKumar1591 committed:
fix for skipping Identity procedure for suci based registration (#6953)

panyogesh committed:
Revert "[5G: MVC][pipelined]Paging Functionality Support For 5G in pipelined (#3535)" (#7160)

This reverts commit f1fbd52f2484d941b7b3385ad211ce0d82f0534d.

Signed-off-by: YOGESH PANDEY <yogesh@wavelabs.ai>

VinayKumar1591 committed:
Fix for guti based registration and T3512 (#7042)

* fix for skipping Identity procedure for suci based registration

* Fix for Guti based registration and T3512

* Addressed review comments

* Memory cleanup for registration procedure

* addressed code review comments

panyogesh committed:
[component: agw] Session Release & UE Deregister (#7029)

Fixes included :
  1. PDU Session Release
  2. UE Deregistration

Testcases :
  1. Unit Testcases of ngap
  2. Testing of PDU Session Release on UERANSIM
  3. Testing of UE Deregistration on UERANSIM
  4. Testing on DSTester

Signed-off-by: YOGESH PANDEY <yogesh@wavelabs.ai>

ganeshg87 committed:
[component:agw] Fix for IA2 changes in AMF (#7175)

Fixes:
  1. Changed the Default algoritm  to E0/IA2
  2. Fixed the offset calculation for mac

Test case:
  1. Tested in UERANSIM
  2. Tested in DSTester

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

Co-authored-by: GANESH IRRINKI <ganesh.irrinki@wavelabs.ai>

electronjoe committed:
[MME] Drop build over-ride to Debug type for ngap sub-task (#7248)

Setting the ngap sub-task to always build Debug variant is incompatible with top level binary build specifications for e.g. production. And therefore cause linking errors when we `make build_oai BUILD_TYPE=RelWithDebInfo`

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

emakeev commented on feg/protos/s6a_proxy.proto:
```diff
@@ -105,6 +105,8 @@ message AuthenticationInformationAnswer {
         bytes xres = 2;
         bytes autn = 3;
         bytes kasme = 4;
+        bytes ck = 5;
+        bytes ik = 6;
```
According to https://www.etsi.org/deliver/etsi_ts/129200_129299/129272/15.10.00_60/ts_129272v151000p.pdf
CK & IK are not a part of E-UTRAN-Vector, they are in UTRAN-Vector (see below) & you can use already defined UTRANVector below.

ulaskozat replied:
Actually, we need a bit more than reusing UTRANVector. I suggest adding an NGRANVector and return kausf (rather than kasme). This would require adding a new field `num_requested_ngran_vectors` to Authentication Information Request so that s6a proxy on subscriberdb knows that it needs to generate `kausf` and not `kasme` for the response. The logic to generate `kausf` needs to be added to subscriberdb. cc @emakeev @panyogesh

emakeev replied:
what RFC/TS defines NGRANVector? S6a proxy converts gRPC fields to diameter AVPs and forwards requests to HSS. Just adding fields to protos won't be enough, we'd need to add the support for them in s6a_proxy & most likely - go-diameter. S6a proxy is dedicated to one specific protocol S6a (29.272), I checked up to v16.6 and I don't see  any mentioning NGRANVector there. s6a_proxy does not generate anything, it just passes through vectors returned by HSS & in general, doesn't depend on or even know of subscriberdb existence.
It's possible that we may need a new service/RPC instead. Currently subscriberdb & s6a_proxy are sharing/implementing the same service and clients decide which one to use: https://github.com/magma/magma/blob/master/lte/gateway/c/oai/lib/s6a_proxy/S6aClient.cpp#L170-L176
But, if protocols are diverging, we may want to consider a new service definition for subscriberdb.

ulaskozat replied:
@emakeev these are 5G extensions there is no HSS etc. Why should proto extensions break FEG?

emakeev replied:
@ulaskozat, they shouldn't break FeG/s6a_proxy, we can add optional fields to AuthenticationInformationRequest/Answer they just won't be supported by FeG. It could be confusing to have 5G extensions defined in s6a_proxy.proto that's why I am suggesting to  consider different service or RPC

ulaskozat replied:
@emakeev I understand the concern. I will ask the team to implement a 5G specific RPC call that can be extended independently.

uri200 replied:
Not sure if you refer to this, but note we already have some 5g fields on s6a_proxy [here](http://github.com/magma/magma/blob/7441d3f1467d142e0f01f101c0e5c0a7a1228592/feg/protos/s6a_proxy.proto#L146-L146) and [here](http://github.com/magma/magma/blob/7441d3f1467d142e0f01f101c0e5c0a7a1228592/feg/protos/s6a_proxy.proto#L281-L281)


Those are on [3gpp 29.272](https://www.etsi.org/deliver/etsi_ts/129200_129299/129272/16.06.00_60/ts_129272v160600p.pdf)

ulaskozat replied:
@uri200 this is a draft PR. Note that 5G fields you are referring to on s6a are for NSA that uses 4G core.

emakeev reviewed (changes requested):
Could you please remove changes to s6a_proxy.proto, they may not be needed (see inline comment).
I'd also suggest splitting this PR & at least have separate PRs for proto changes

ulaskozat commented on .circleci/config.yml:
```diff
@@ -17,6 +17,10 @@ aliases:
     paths: "orc8r lte"
   - &c_cpp_build_verify
     paths: "orc8r/gateway/c orc8r/protos lte/gateway/c lte/protos"
+  - &session_manager_build_verify
```
why does this PR touch circleci directory?

ulaskozat commented on orc8r/cloud/deploy/bare-metal-ansible/deploy.sh:
```diff
@@ -20,7 +20,8 @@ cd kubespray
 git checkout $KUBESPRAY_RELEASE
 # Copy ``inventory/sample`` as ``inventory/$CLUSTER_NAME``
-cp -rnp inventory/sample inventory/$CLUSTER_NAME
+mkdir -p ../inventory/$CLUSTER_NAME
+cp -rnp inventory/sample ../inventory/$CLUSTER_NAME
```
why are these touched?

themarwhal commented on lte/gateway/c/session_manager/EnumToString.cpp:
```diff
@@ -213,4 +215,22 @@ std::string event_trigger_to_str(EventTrigger event_trigger) {
   }
 }
 
+std::string request_origin_type_to_str(
```
Is this a new change? I see it already on master: https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/c/session_manager/EnumToString.cpp

themarwhal reviewed (changes requested):
This PR seems to contain some changes that already exist on master. Can you rebase and try to apply the changes again?

themarwhal committed:
[component:agw] Handling of the UEContextRequest present information in UE and populating in AMF (#7271)

Fixes:
 1. NGAP decodes UEContextRequest IE received from InitialUEMessage and Passes to AMF thread
 2. AMF will trigger InitialContextSetupRequest based on UEContextRequest parameter

Test case:
 1. Tested Registration with UEContextRequest IE  using UERANSIM
 2. Tested Registration without UEContextRequest IE  using UERANSIM

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

Co-authored-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

themarwhal committed:
[component:agw] Handling AUTH Failure with Reason #71(ngKSI is already in use) (#7272)

Fix:
  1. In UE rejects AUTH failure Response with Reason #71, AMF reinitiate AUTH with
      restart AUTH procedure with new ngKSI
Test case :
1. Tested Registration  using UT/(ueRanSImulator)

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

Co-authored-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

themarwhal committed:
Backport gnutls host change to unblock CI (#7273)

Signed-off-by: Marie Bremner <marwhal@fb.com>

ulaskozat committed:
Fix RYU Postinstall Patch Apply (#7274) (#7275)

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

Co-authored-by: Mykola Yurchenko <koolzz@fb.com>

VinayKumar1591 committed:
Auth failure changes (#7333)

* 5G Authentication failure changes
1. MAC FAILURE handling changes
2. SYNCH AILURE handling changes

* 5G Authentication failure changes
1. UT FIXES

* 5G Authentication failure changes
1.Addressed code revie comments

ganeshg87 committed:
Enforcement functionality support on magma-5g

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

prabinakpattnaik committed:
Enforcement functionality support on magma-5g (#7492)

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

Co-authored-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

tmdzk committed:
[CI] disable integ-test (#7285)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

panyogesh committed:
[component: agw] Changes to support Sync Failure (#7439)

Fixes :
  1. Create new proto file to send message to Subscriber
  2. Create the wrapper for sending grpc messages messages
  3. New handler for Subscriber Auth response in AMF
  4. Support for Sync Failure

Testing :
  1. Basic test on Simulator
  2. Related scenarios verified on TerraVM

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

panyogesh committed:
[component:agw] Fix for NAS Decode failure on GUTI (#7550)

Fixes:
  1. Fix for NAS decode failure on getting GUTI request
     from TerraVM simulator.
  2. Fix for LTE test failures.

Testing:
  1. Tested basic flow with simulator.
  2. Run unit test of ngap and amf.

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

VinayKumar1591 committed:
Fix for XRES_STAR Derivation (#7551)

* Fix for XRES* derivation and validation of RES*,XRES*

* Fix for XRES_STAR Derivation

1. Addressed review comments

* Fix for XRES_STAR Derivation

1. Addressed review comments and added spec reference

prabinakpattnaik committed:
Multi Gnbs support in SMF/UPF

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

prabinakpattnaik committed:
Revert "Multi Gnbs support in SMF/UPF"

This reverts commit 196011543ae484b790313befa829ab504387a16d.

prabinakpattnaik committed:
Merge branch 'magma-5G' into magma-5G

prabinakpattnaik committed:
Multi gNB support (#7602)

* Multi Gnbs support in SMF and UPF

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

* Multi Gnbs support in SMF and UPF

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

prabinakpattnaik committed:
Paging functionality support in magma-5G (#7606)

Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

prabinakpattnaik committed:
Revert "Paging functionality support in magma-5G" (#7622)

This reverts commit 1b88641ba506a2c2530b27dee8a4aadc6e6a2dda.

panyogesh committed:
[component:agw] GRPC tuning between AMF & SMF (#7623)

Fix:
  1. Streamline SMF Client Services.
  2. Cleaned up SMF GRPC client calls from AMF.

Test:
  1. Tested using Simulator.

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

VinayKumar1591 committed:
Fix for the Pdu Session Establishment Issue (#7626)

1. imsi corrupted while generating security keys
  and  invalid imsi sent to mobilityd
2. corrected imsi64 to imsi string conversion

ulaskozat committed:
Fix pylint error (#7632)

Signed-off-by: Ulas Kozat <kozat@fb.com>

tmdzk committed:
[CI] Remove c-cpp-codecov from 5G branch

Signed-off-by: Timothée Dzik <timdzik@fb.com>

panyogesh committed:
[component:agw] UE Data clean up in Deregister (#7661)

Fix:
  1. Remove UE from the hash table during de-register
  2. Comment out the monitor thread in Pipelined.
     Reason being SMF needs the handler and some cleanup in UPF

Test:
  1. Tested in Simulator

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

ganeshg87 committed:
[component:agw] SUCI/IMSI based Registration fails when IMSI's MNC is sent as 3 DIGIT #7619 (#7666)

Fix:
       1. Resolved plmn, imsi64 while accessing 3digit mnc value from imsi in amf_context

    Testcases:
       1. Registration with 2 digit MNC successfull with UERANSIM
       1. Registration with 3 digit MNC successfull with UERANSIM

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

ulaskozat committed:
Add logs and disable xres check (#7675)

Signed-off-by: Ulas Kozat <kozat@fb.com>

ulaskozat committed:
Fixed miscomputed snn and reenabled xres* check (#7676)

Signed-off-by: Ulas Kozat <kozat@fb.com>

tmdzk committed:
[CI]Push to right folder for 5g branch (#7696)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

Kaleem-Wavelabs committed:
Fixes:
1) Added the missing code functionality to support T3560 timer

Test cases:
1) Run with UERANSIM to miss the response for authenticatoin request
2) Run with UERANSIM to send the authentication success and failure messages after initial failures

Signed-off-by: Kaleemullah Mohammed <kaleemullah.mohammed@wavelabs.ai>

panyogesh committed:
[component:agw] Mobilityd IP Release fix (#7697)

Issue:
  1. Once the session is released IP addresse
     needs to be freed in mobilityD

Fix:
  1. Enhanced the M5G mobility Service client to support
     IP address release message to MobolityD
  2. Done some changes in IP Address allocation part
     to store the information in SMF context.

Test:
  1. Tested on UERANSIM and verified logs of mobilityd

Log : Released IP 192.168.128.223 for sid IMSI901700000000001

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

Kaleem-Wavelabs committed:
Periodic Registration fixes and timer changes (#7809)

Signed-off-by: Kaleemullah Mohammed <kaleemullah.mohammed@wavelabs.ai>

panyogesh committed:
[component: agw]: Periodic Registartion fixes (#8045)

Fixes:
  1. Support for Periodic Registration
  2. Support for timer 3550
  3. Fixes for GUTI encoding/decoding
  4. Fixes for Context lookup based on GUTI
  5. Minor Clean ups
  6. NGKSI Fixes

Test Covered :
  1. Periodic Registration Idle Mode/Active Mode
  2. All P0 related testcases
  3. Basic Registration/De-Registration in loop
  4. SIT issue for ngKSI
  5. Validated in TerraVM and UERANSIM setup

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

uri200 commented on lte/gateway/c/session_manager/test/test_local_enforcer.cpp:
```diff
@@ -2686,15 +2704,9 @@ TEST_F(LocalEnforcerTest, test_final_unit_redirect_activation_and_termination) {
-      .Times(1);
+      *pipelined_client, deactivate_flows_for_rules_for_termination(
+                             IMSI1, ip_addr, ipv6_addr, CheckTeids(teids),
+                             RequestOriginType::WILDCARD));
```
why are we using wildcard? We do want to check there is one of eacn (Gx and Gy)

ekowtaylor commented:
@Kaleem-Wavelabs, this PR has now gone stale. To maintain a healthy codebase, it will be automatically closed unless there are plans to rebase and merge by close of week

ekowtaylor commented:
@Kaleem-Wavelabs, this PR has now gone stale. To maintain a healthy codebase, it will be automatically closed unless there are plans to rebase and merge by close of week

panyogesh commented:
Closing the pull request as no longer needed.
